### PR TITLE
Add primary key to decoderbuf

### PR DESCRIPTION
**Pull request by gburanov**

### CODE REVIEW

File: src/decoderbufs.c [hunk 1]
```diff
@@ -595,6 +595,25 @@ static int tuple_to_tuple_msg(Decoderbufs__DatumMessage **tmsg,
     /* set the column name */
     datum_msg.column_name = NameStr(attr->attname);
 
+    /* set is indxed */
+      if (indexT != NULL) {
```
Comment (itszootime) [inline src/decoderbufs.c]: Indentation is wrong here

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+sudo: required
+
+language: generic
+
+services:
+  - docker
+
+before_install:
+  - docker build  .
```
Comment (itszootime) [inline .travis.yml:9]: Is there any more value here than checking if the plugin compiles? If that's all there is, I don't think it's worth the 30 minute Travis build...

File: src/decoderbufs.c [hunk 3]
```diff
@@ -571,7 +571,7 @@ static void set_datum_value(Decoderbufs__DatumMessage *datum_msg, Oid typid,
 /* convert a PG tuple to an array of DatumMessage(s) */
 static int tuple_to_tuple_msg(Decoderbufs__DatumMessage **tmsg,
                                Relation relation, HeapTuple tuple,
-                               TupleDesc tupdesc) {
+                               TupleDesc tupdesc, TupleDesc indexT) {
```
Comment (itszootime) [inline src/decoderbufs.c]: `index_tupledesc` for consistency?

File: src/decoderbufs.c [hunk 4]
```diff
@@ -595,6 +595,25 @@ static int tuple_to_tuple_msg(Decoderbufs__DatumMessage **tmsg,
     /* set the column name */
     datum_msg.column_name = NameStr(attr->attname);
 
+    /* set is indxed */
```
Comment (itszootime) [inline src/decoderbufs.c]: `check if this attr is part of the index`?

File: src/decoderbufs.c [hunk 5]
```diff
@@ -595,6 +595,25 @@ static int tuple_to_tuple_msg(Decoderbufs__DatumMessage **tmsg,
...
+      if (indexT != NULL) {
+        int		j;
+        bool	found_col = false;
```
Comment (itszootime) [inline src/decoderbufs.c]: `is_indexed`?

File: src/decoderbufs.c [hunk 6]
```diff
@@ -648,6 +667,13 @@ static void pg_decode_change(LogicalDecodingContext *ctx, ReorderBufferTXN *txn,
                            !OidIsValid(relation->rd_replidindex)));
 
   /* set common fields */
+  // set primary key only if it exists for the table
+  TupleDesc indexT = NULL;
```
Comment (itszootime) [inline src/decoderbufs.c]: `index_tupdesc` again?

File: src/decoderbufs.c [hunk 7]
```diff
@@ -648,6 +667,13 @@ static void pg_decode_change(LogicalDecodingContext *ctx, ReorderBufferTXN *txn,
...
+  Relation indexRel;
```
Comment (itszootime) [inline src/decoderbufs.c]: `index_rel` - snake case consistency?

File: src/decoderbufs.c [hunk 8]
```diff
@@ -702,14 +728,18 @@ static void pg_decode_change(LogicalDecodingContext *ctx, ReorderBufferTXN *txn,
...
-                           &change->data.tp.oldtuple->tuple, tupdesc);
+                           &change->data.tp.oldtuple->tuple, tupdesc, NULL);
       }
       break;
     default:
       Assert(0);
       break;
   }
 
+  if (indexT != NULL) {
```
Comment (itszootime) [inline src/decoderbufs.c]: No behavioural difference, but easier to follow: `if (index_rel != NULL)`

File: proto/pg_logicaldec.proto [hunk 9]
```diff
@@ -16,6 +16,7 @@ message Point {
 message DatumMessage {
 	string column_name = 1;
 	int64 column_type = 2;
+	bool is_pk_indexed = 12;
```
Comment (itszootime) [inline proto/pg_logicaldec.proto]: Maybe just `primary_key` or `is_primary_key`?

**Review by itszootime — COMMENTED** [review]
Nice job on this one 🙂. My comments are mostly trivial naming suggestions.

File: .travis.yml (hunk 2 above)
Comment (gburanov) [reply .travis.yml:9]: I added threading warnings as errors, so we can catch some warnings in prs as well now )

File: src/decoderbufs.c (hunk 3 above)
Comment (gburanov) [reply src/decoderbufs.c]: done

File: src/decoderbufs.c (hunk 4 above)
Comment (gburanov) [reply src/decoderbufs.c]: done

File: src/decoderbufs.c (hunk 5 above)
Comment (gburanov) [reply src/decoderbufs.c]: 👍

File: proto/pg_logicaldec.proto (hunk 9 above)
Comment (gburanov) [reply proto/pg_logicaldec.proto]: 👍

File: src/decoderbufs.c (hunk 8 above)
Comment (gburanov) [reply src/decoderbufs.c]: 👍

**Comment by gburanov** [comment]
I made the changes ) Plz recheck @itszootime 
If u will approve, I will ask @dynamix  on how to deploy it

**Review by itszootime — APPROVED** [review]

**Pull request merged**